### PR TITLE
[feat]: [PIE-4876]: Trigger Drawer Listing Call

### PR DESCRIPTION
**Pull request by sribalij**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38082)

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java` line 440 at r1](https://reviewable.io/reviews/harness/harness-core/38082#-ND5dWqrDXtcYihsmr6h:-ND5dWqrDXtcYihsmr6i:bb7phvt) ([raw file](https://github.com/harness/harness-core/blob/eb8a162e81ba5ff1966c77b49869dea52c5924bc/956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java#L440)):*
> ```Java
>   PL_NO_EMAIL_FOR_SAML_ACCOUNT_INVITES("No email for users in account where SAML auth is enabled", HarnessTeam.PL),
>   PL_ENABLE_GOOGLE_SECRET_MANAGER_IN_NG("Enable Google Secret Manager in NG", HarnessTeam.PL),
>   CD_TRIGGER_CATALOG("Enables UI for Trigger cataloh for Nexus ", HarnessTeam.CDC);
> ```

typo
___
*[`pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/entity/metadata/catalog/TriggerCatalogType.java` line 37 at r1](https://reviewable.io/reviews/harness/harness-core/38082#-ND5e7Qf8DUFG-pZYXgs:-ND5e7Qf8DUFG-pZYXgt:b-d2m7tv) ([raw file](https://github.com/harness/harness-core/blob/eb8a162e81ba5ff1966c77b49869dea52c5924bc/pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/entity/metadata/catalog/TriggerCatalogType.java#L37)):*
> ```Java
> 
>   // Manifest
>   @JsonProperty("HelmChart") HELM_CHART("HelmChart"),
> ```

discussed offline



_Code quote:_
```Java
ate final String displayName;
```

File: pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/dto/NGTriggerCatalogDTO.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ApiModel("TriggerCatalogResponse")
+@Schema(name = "TriggerCatalogResponse", description = "This has details of the retrieved Trigger Catalog.")
+public class NGTriggerCatalogDTO {
+  @Schema(description = ConnectorConstants.CONNECTOR_CATALOGUE_LIST) List<TriggerCatalogItem> catalog;
```
Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/dto/NGTriggerCatalogDTO.java:33]: wrong schema?

Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/dto/NGTriggerCatalogDTO.java:33]: incorrect scgema

File: pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/resource/NGTriggerResource.java [hunk 2]
```diff
@@ -248,4 +251,18 @@ public interface NGTriggerResource {
...
+
+  @GET
+  @Path("catalog")
+  @ApiOperation(value = "Get Trigger catalog", nickname = "getTriggerCatalog")
+  @Operation(operationId = "getTriggerCatalog", summary = "Lists all Triggers",
+      description = "Lists all the Triggers for the given Account ID.",
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(responseCode = "default", description = "Returns the Connector catalogue response")
```
Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/resource/NGTriggerResource.java:263]: connector

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerCatalogHelper.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.ejb.Singleton;
+
+@Singleton
+public class TriggerCatalogHelper {
+  @Inject TriggerCatalogFilter enumFilter;
+
+  public List<TriggerCatalogItem> getConnectorTypeToCategoryMapping(String accountIdentifier) {
```
Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerCatalogHelper.java:28]: wrong method name

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java [hunk 4]
```diff
@@ -945,4 +948,8 @@ public Object fetchExecutionSummaryV2(String planExecutionId, String accountId,
     return NGRestUtils.getResponse(
         pipelineServiceClient.getExecutionDetailV2(planExecutionId, accountId, orgId, projectId));
   }
+  @Override
+  public List<TriggerCatalogItem> getTriggerCatalog(String accountIdentifier) {
+    return triggerCatalogHelper.getConnectorTypeToCategoryMapping(accountIdentifier);
```
Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:953]: wrong method name

**Comment by sribalij** [comment]
trigger codeformat

**Review by sribalij — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: 13 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/dto/NGTriggerCatalogDTO.java` line 33 at r2](https://reviewable.io/reviews/harness/harness-core/38082#-ND4WDY2SAQ2jVdb30Fn-r2-33:-NDAXUM1CdrWqXPlYWOx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0cfde8a79059023546a95c57837318a253070bfc/pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/dto/NGTriggerCatalogDTO.java#L33)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

incorrect scgema
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/entity/metadata/catalog/TriggerCatalogType.java` line 37 at r1](https://reviewable.io/reviews/harness/harness-core/38082#-ND5e7Qf8DUFG-pZYXgs:-NDAXVfY8j9K_EeL_0gM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/eb8a162e81ba5ff1966c77b49869dea52c5924bc/pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/beans/entity/metadata/catalog/TriggerCatalogType.java#L37)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

discussed offline
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/resource/NGTriggerResource.java` line 263 at r2](https://reviewable.io/reviews/harness/harness-core/38082#-ND4WDY2SAQ2jVdb30Fr-r2-263:-NDAXWu_8NgtP7SzAI36:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0cfde8a79059023546a95c57837318a253070bfc/pipeline-service/modules/ng-triggers/contracts/src/main/java/io/harness/ngtriggers/resource/NGTriggerResource.java#L263)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

connector
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerCatalogHelper.java` line 28 at r2](https://reviewable.io/reviews/harness/harness-core/38082#-ND4WDY2SAQ2jVdb30Fu-r2-28:-NDAXY9l6aX1CiDRgrBr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0cfde8a79059023546a95c57837318a253070bfc/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/helpers/TriggerCatalogHelper.java#L28)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

wrong method name
</blockquote></details>

Done.
___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java` line 953 at r2](https://reviewable.io/reviews/harness/harness-core/38082#-ND4WDY2SAQ2jVdb30Fy-r2-953:-NDAXYvhAAVBvUjC3eEs:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0cfde8a79059023546a95c57837318a253070bfc/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java#L953)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

wrong method name
</blockquote></details>

Done.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 7 of 14 files at r1, 2 of 4 files at r2, 3 of 5 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: 6 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained.

**Comment by sribalij** [comment]
trigger codebasehashcheck

**Review by sahilhindwani-harness — DISMISSED** [review]
> Reviewed 7 of 14 files at r1, 2 of 4 files at r2, 3 of 5 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: 1 change requests. 5. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 2 of 5 files at r3, 1 of 2 files at r4, 3 of 9 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 3 of 9 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: :shipit: complete! ✓

**Review by praveensugavanam — APPROVED** [review]
> Reviewed 6 of 14 files at r1, 2 of 4 files at r2, 2 of 5 files at r3, 1 of 2 files at r4, 3 of 9 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38082)* status: :shipit: complete! ✓

**Pull request merged**